test/functional/lua/runtime_spec.lua

```diff
@@ -0,0 +1,224 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local eq = helpers.eq
+local eval = helpers.eval
+local exec = helpers.exec
+local mkdir_p = helpers.mkdir_p
+local rmdir = helpers.rmdir
+local write_file = helpers.write_file
+
+describe('runtime:', function()
+  local xhome = 'Xhome'
+  local pathsep = helpers.get_pathsep()
+  local xconfig = xhome .. pathsep .. 'Xconfig'
+
+  before_each(function()
+    clear()
+    mkdir_p(xconfig .. pathsep .. 'nvim')
+  end)
+
+  after_each(function()
+    rmdir(xhome)
+  end)
+
+  describe('plugin', function()
+    before_each(clear)
+    it('loads plugin/*.lua from XDG config home', function()
+      local plugin_folder_path = table.concat({xconfig, 'nvim', 'plugin'}, pathsep)
+      local plugin_file_path = table.concat({plugin_folder_path, 'plugin.lua'}, pathsep)
+      mkdir_p(plugin_folder_path)
+      write_file(plugin_file_path, [[ vim.g.lua_plugin = 1 ]])
+
+      clear{ args_rm={'-u' }, env={ XDG_CONFIG_HOME=xconfig }}
```

Comment:
I don't like using `clear` here to set a new nvim instance in the middle of a test -- it's a bit weird that `mkdir_p` communicates with the firts nvim, then you clear, and then now it's a new nvim as the session.

Reply:
double clear is kind of weird true . But the test that check for stuff happening at startup ( plugin & ftdetect here) need the second clear . The rest don't . I've removed it from them .